graalvm/generate-config.sh

```diff
@@ -6,34 +6,48 @@ if [ -z "$GRAALVM_HOME" ]; then
 fi
 
 SUITE=./graalvm/suite.yml
-while getopts ":hs:" opt; do
+TARGET_DIR=./target
+while getopts ":hs:t:" opt; do
   case ${opt} in
     h)
       echo "Usage: $0 [-h] [-s <SUITE>]"
       echo "  -h:         Display this help message."
       echo "  -s <SUITE>: The path to the suite.yml file. Default: ./graalvm/suite.yml"
+      echo "  -t <TARGET_DIR>: The location of the maven target directory. Default: ./target/"
       exit 0
       ;;
+    t)
+      TARGET_DIR=$OPTARG
+      ;;
     s)
       SUITE=$OPTARG
       ;;
     ?)
-      echo "Invalid option: $OPTARG" 1>&2
+      echo "Invalid option: ${opt}" 1>&2
       exit 1
       ;;
   esac
 done
 
-if [ ! -f ./target/iguana-4.0.0.jar ]; then
+if [ ! -f "$TARGET_DIR"/iguana.jar ]; then
   mvn -DskipTests package
 fi
 
-"$GRAALVM_HOME"/bin/java -agentlib:native-image-agent=config-output-dir=./src/main/resources/META-INF/native-image -jar ./target/iguana-4.0.0.jar --help      > /dev/null
-"$GRAALVM_HOME"/bin/java -agentlib:native-image-agent=config-merge-dir=./src/main/resources/META-INF/native-image -jar ./target/iguana-4.0.0.jar -is "$SUITE" > /dev/null
-"$GRAALVM_HOME"/bin/java -agentlib:native-image-agent=config-merge-dir=./src/main/resources/META-INF/native-image -jar ./target/iguana-4.0.0.jar "$SUITE"     > /dev/null
+if [ ! -d src/main/resources/META-INF/native-image/ ]; then
+  mkdir -p src/main/resources/META-INF/native-image/
+fi
+
+# Move generated configuration files from tests to the resources
+if [ -f "$TARGET_DIR"/native/agent-output/test/resource-config.json ]; then
+  mv "$TARGET_DIR"/native/agent-output/test/* src/main/resources/META-INF/native-image/
+fi
+
+"$GRAALVM_HOME"/bin/java -agentlib:native-image-agent=config-merge-dir=src/main/resources/META-INF/native-image/ -jar "$TARGET_DIR"/iguana.jar --help      > /dev/null
+"$GRAALVM_HOME"/bin/java -agentlib:native-image-agent=config-merge-dir=src/main/resources/META-INF/native-image/ -jar "$TARGET_DIR"/iguana.jar -is "$SUITE" > /dev/null
+"$GRAALVM_HOME"/bin/java -agentlib:native-image-agent=config-merge-dir=src/main/resources/META-INF/native-image/ -jar "$TARGET_DIR"/iguana.jar "$SUITE"     > /dev/null
```

Comment:
If I get this right, you are running all possible combinations of commandline arguments. Is that right? Please add a comment that references the section of the GraalVM docu that explains why this is necessary. (And add, if further helpful, an additional comment of your own)

Reply:
I don't think the documentation explicitly says, that this is necessary. I just couldn't figure out how to configure this manually to be honest, so I just implemented it this way. 
Implementing it this way might also be better, because the configurations don't need to be manually updated, everytime something relevant in the code changes.

Reply:
ok